src/js/control-bar/playback-rate-menu/playback-rate-menu-button.js

```diff
@@ -0,0 +1,108 @@
+import MenuButton from '../../menu/menu-button.js';
+import Menu from '../../menu/menu.js';
+import PlaybackRateMenuItem from './playback-rate-menu-item.js';
+import * as Lib from '../../lib.js';
+
+/**
+ * The component for controlling the playback rate
+ *
+ * @param {vjs.Player|Object} player
+ * @param {Object=} options
+ * @constructor
+ */
+class PlaybackRateMenuButton extends MenuButton {
+
+  constructor(player, options){
+    super(player, options);
+
+    this.updateVisibility();
+    this.updateLabel();
+
+    this.on(player, 'loadstart', this.updateVisibility);
+    this.on(player, 'ratechange', this.updateLabel);
+  }
+
+  createEl() {
+    let el = MenuButton.prototype.createEl.call(this);
+
+    this.labelEl_ = Lib.createEl('div', {
+      className: 'vjs-playback-rate-value',
+      innerHTML: 1.0
+    });
+
+    el.appendChild(this.labelEl_);
+
+    return el;
+  }
+
+  // Menu creation
+  createMenu() {
+    let menu = new Menu(this.player());
+    let rates = this.player().options()['playbackRates'];
+
+    if (rates) {
+      for (let i = rates.length - 1; i >= 0; i--) {
+        menu.addChild(
+          new PlaybackRateMenuItem(this.player(), { 'rate': rates[i] + 'x'})
+        );
+      }
+    }
+
+    return menu;
+  }
+
+  updateARIAAttributes() {
+    // Current playback rate
+    this.el().setAttribute('aria-valuenow', this.player().playbackRate());
+  }
+
+  onClick() {
+    // select next rate option
+    let currentRate = this.player().playbackRate();
```

Comment:
We definitely need to be more consistent. We use `this.player_` and `this.player()` seemingly at random.